src/v/cloud_topics/batcher/aggregator.cc

```diff
@@ -51,47 +49,32 @@ namespace {
 /// concatenating multiple chunks the offset has to be corrected.
 /// This is done using the `base_byte_offset` parameter.
 template<class Clock>
-void make_dl_placeholder_batches(
-  prepared_placeholder_batches<Clock>& ctx,
+void make_dl_placeholders(
+  prepared_extents<Clock>& ctx,
   core::write_request<Clock>& req,
   const core::serialized_chunk& chunk) {
-    auto result = std::make_unique<batches_for_req<Clock>>();
-    for (const auto& b : chunk.batches) {
-        dl_placeholder placeholder{
+    auto result = std::make_unique<extents_for_req<Clock>>();
+    for (const auto& b : chunk.extents) {
+        extent_meta placeholder{
           .id = ctx.id,
-          .offset = first_byte_offset_t(ctx.size_bytes),
-          .size_bytes = byte_range_size_t(b.size_bytes),
+          .first_byte_offset = first_byte_offset_t(ctx.size_bytes),
+          .byte_range_size = byte_range_size_t(b.byte_range_size),
+          .base_offset = b.base_offset,
+          .committed_offset = b.committed_offset,
         };
 
-        storage::record_batch_builder builder(
-          model::record_batch_type::dl_placeholder, b.base);
-
-        // TX data (producer id, control flag) are not copied from 'src' yet.
-
-        // Put the payload
-        builder.add_raw_kv(
-          serde::to_iobuf(dl_placeholder_record_key::payload),
-          serde::to_iobuf(placeholder));
-
-        // TODO: fix this
-        for (int i = 1; i < b.num_records - 1; i++) {
-            iobuf empty;
-            builder.add_raw_kv(
-              serde::to_iobuf(dl_placeholder_record_key::empty),
-              std::move(empty));
-        }
-        result->placeholders.push_back(std::move(builder).build());
-        ctx.size_bytes += b.size_bytes;
+        result->extents.emplace_back(placeholder);
```

Comment:
it would be helpful to have one sentence in the commit message, or a review comment here: going from serialized batch to unserialized extent_meta feels like a non-obvious change?